rust/src/modbus/detect.rs

```diff
@@ -0,0 +1,1345 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use super::modbus::ModbusTransaction;
+use lazy_static::lazy_static;
+use regex::Regex;
+use sawp_modbus::{AccessType, CodeCategory, Data, FunctionCode, Message};
+use std::ffi::CStr;
+use std::ops::{Range, RangeInclusive};
+use std::os::raw::{c_char, c_void};
+use std::str::FromStr;
+
+lazy_static! {
+    static ref ACCESS_RE: Regex = Regex::new("^\\s*\"?\\s*access\\s*(read|write)\\s*(discretes|coils|input|holding)?(?:,\\s*address\\s+([<>]?\\d+)(?:<>(\\d+))?(?:,\\s*value\\s+([<>]?\\d+)(?:<>(\\d+))?)?)?\\s*\"?\\s*$").unwrap();
```

Comment:
Could we use nom instead of regexes to parse the rule keyword value ?

Reply:
I think @jasonish (or what it @chifflier?) was saying that for string parsing `nom` isn't the best choice. Not sure if that is still the consensus?

Reply:
What would be the advantage of using `nom` over `regex`? The regex is a port of the pcre regex in the prior C module.

`nom` does support `&str` as an input instead of bytes IIRC. 

I can only find [this example](https://github.com/OISF/suricata/blob/master/rust/src/asn1/parse_rules.rs#L124) from asn1 keyword parsing.

Reply:
I found using nom was ok for parsing a rule, so its not that bad.  I think it comes down to whatever is simpler to maintain..  For simple stuff a regex might be simpler, but we do have some regex that are really hard to maintain when changes are needed, and those might be better with nom, or even simple word splitting and parsing.

Reply:
> What would be the advantage of using nom over regex? 

It seems more readable/maintainable to me
So, this is subjective, and I am good either way.

> I can only find this example from asn1 keyword parsing.

Another example is `rs_http2_detect_settingsctx_parse`